### PR TITLE
Adding proxy for ondemand repo.

### PR DESCRIPTION
**Pull request by pedmon**
We have a secure environment that requires proxy to get out for dnf repos.  This adds the option to set a proxy for the OnDemand repo.

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -233,6 +235,7 @@
   $repo_baseurl_prefix = 'https://yum.osc.edu/ondemand',
   Variant[Stdlib::HTTPSUrl, Stdlib::HTTPUrl, Stdlib::Absolutepath]
   $repo_gpgkey = 'https://yum.osc.edu/ondemand/RPM-GPG-KEY-ondemand-SHA512',
+  String $repo_proxy = '_none_',
```
Comment (treydock) [inline manifests/init.pp:238]:
```suggestion
  Optional[String[1]] $repo_proxy = undef,
```

**Comment by treydock** [comment]
@pedmon I left inline suggestion.  Generally you want default values to be `undef` if the parameter is optional.

**Comment by pedmon** [comment]
Cool. I've accepted that suggestion.

**Comment by treydock** [comment]
@pedmon Can you rebase against latest master branch? I pushed some changes that should resolve failing tests.

**Comment by treydock** [comment]
This is being released with v4.0.0 once Github Actions complete

**Pull request merged**